### DISCUSSION TITLE
Next Build generating an empty CSS file

### DISCUSSION THREAD

**Discussion by russellhill** [vercel/next.js#53905] [Help]
### Summary

**Using Next.js 13.1.6**

Hi, hope someone can help me diagnose this...

Our Next.js build process is generating one empty CSS chunk file (amongst many others that are not empty). The CSS/JS chunks are hosted on our CDN, sometimes the web pages that use components that reference this empty CSS file crash with 500 error.

The 500 error comes from the Azure CDN, so clearly that has some issue with zero byte files, but this wouldn't be a problem if the file was not generated in the first place.

I can see from the react-loadable-manifest.json that there are 7 JS components that reference this empty CSS file. These are all dynamically loaded components loaded using `dynamic()`. But we have a total of 24 dynamically loaded JS components yet only these 7 show in the manifest as using the empty CSS file.

Our list of dynamic components:

`import dynamic from "next/dynamic";

const DynamicComponents = {
    pageHero: dynamic(() => import("../components/PageHero/PageHero")),
    image: dynamic(() => import("../components/FullImage/FullImage")),
    simplePanels: dynamic(() => import("../components/SimplePanels/SimplePanels")),
    quotePanel: dynamic(() => import("../components/QuotePanel/QuotePanel")),
    simpleList: dynamic(() => import("../components/SimpleList/SimpleList")),
    CtaPanel: dynamic(() => import("../components/CtaPanel/CtaPanel")),
    newsList: dynamic(() => import("../components/NewsList/NewsList")),
    fullWidthHeading: dynamic(() => import("../components/FullWidthHeading/FullWidthHeading")),
    featurePanel: dynamic(() => import("../components/FeaturePanel/FeaturePanel")),
    imageWithCallout: dynamic(() => import("../components/ImageWithCallout/ImageWithCallout")),
    newsList: dynamic(() => import("../components/NewsList/NewsList")),
    gallery: dynamic(() => import("../components/Gallery/Gallery")),
    relatedNews: dynamic(() => import("../components/NewsArticle/RelatedNews")),
    videoPlayer: dynamic(() => import("../components/VideoPlayer/VideoPlayer")),
    imageWithCalloutList: dynamic(() => import("../components/ImageWithCalloutList/ImageWithCalloutList")),
    staffCarousel: dynamic(() => import("../components/StaffCarousel/StaffCarousel")),
    imageCarousel: dynamic(() => import("../components/ImageCarousel/ImageCarousel")),
    headingBlock: dynamic(() => import("../components/HeadingBlock/HeadingBlock")),
    contactForm: dynamic(() => import("../components/ContactForm/ContactForm")),
    textBlock: dynamic(() => import("../components/TextBlock/TextBlock")),
    CtaBannerPanel: dynamic(() => import("../components/CtaBannerPanel/CtaBannerPanel")),
    linkList: dynamic(() => import("../components/LinkList/LinkList")),
    imageLink: dynamic(() => import("../components/ImageLink/ImageLink")),
    gameRating: dynamic(() => import("../components/GameRating/GameRating")),
};

export default DynamicComponents;` 

Of these only the following 7 reference the empty CSS:

`FullImage
FullWidthHeading
GameRating
HeadingBlock
ImageLink
LinkList
QuotePanel`

I cannot see any reason why these 7 reference the empty CSS file and the remaining ones don't.

Sample from the react-loadable-manifest.json:

`  "..\\helpers\\DynamicComponents.js -> ../components/FullImage/FullImage": {
    "id": 7964,
    "files": [
      "static/css/ef46db3751d8e999.css",
      "static/chunks/964.5332531e9def591d.js"
    ]
  },
`

The empty CSS file chunk name is `static/css/ef46db3751d8e999.css`

Any ideas on this?

If I could track down what was causing the empty CSS chunk to be generated, I think that'd help a lot, but I can't seem to see anything in the manifest files or using @next/bundle-analyzer that helps.

**Comment by janel-ms** [comment]
The exact same empty css file is created for my app - would love to have some insight about this. I'm using Next 14.
